### PR TITLE
ARROW-14050: [C++] Make TDigest/Quantile kernels return nulls instead

### PR DESCRIPTION
**Pull request by lidavidm**
As requested, this makes it so that TDigest/Quantile return nulls instead of empty arrays, making them easier to consume from bindings, and making them compose better with kernels like list_flatten and list_element.

### CODE REVIEW

File: cpp/src/arrow/compute/kernels/aggregate_test.cc [hunk 1]
```diff
@@ -3312,6 +3313,71 @@ TEST_F(TestRandomFloatQuantileKernel, Sliced) {
...
+    ValidateOutput(*out_array);
+    AssertArraysEqual(*ArrayFromJSON(float64(), "[null]"), *out_array, /*verbose=*/true);
+  }
+}
+
+TEST(TestQuantileKernel, Scalar) {
+  for (const auto& ty : {float64(), int64(), uint64()}) {
+    QuantileOptions options(std::vector<double>{0.0, 0.5, 1.0});
+    EXPECT_THAT(Quantile(*MakeScalar(ty, 1), options),
+                ResultWith(ArrayFromJSON(float64(), "[1.0, 1.0, 1.0]")));
```
Comment (pitrou) [inline cpp/src/arrow/compute/kernels/aggregate_test.cc:3338]: Did you mean to also test with a null scalar here?

**Review by pitrou — APPROVED** [review]
LGTM. Just one question.

File: cpp/src/arrow/compute/kernels/aggregate_test.cc (hunk 1 above)
Comment (lidavidm) [reply cpp/src/arrow/compute/kernels/aggregate_test.cc:3338]: Yes, added (also added to TDigest's tests)

**Comment by ianmcook** [comment]
Works great from the R bindings. Thanks!

**Review by ianmcook — APPROVED** [review]

**Pull request closed without merging**